### ISSUE TITLE
Order of exec command ran inside scenario

### ISSUE THREAD

**Issue by Jean-PhilippeD** [GladysAssistant/Gladys#137]
Hello,
When chaining device exec in launcher (scenario), does the command are followed "inside promises" ?

Because when I run a philips Hue "set-on" command followed by "set-brightness", it's like the "set-brightness" is trying to be set before the device is "on", so the hue command failed and put back the device off.

**Comment by Pierre-Gilles** [comment]
Nope, it's a promise.map done so all tasks are running in parallel for maximum performance !

For your problem, it's really interesting.. for the moment maybe create just a script and chain all promises :)

But we should think about another way of specifying an order.

**Comment by Jean-PhilippeD** [comment]
That's what I did, I've created a script. 

But I'm not totally sure about the error I've described in fact..

Because, a few days ago , I have modified another script to add a "turn off light 5 min after turning on".
But I made a mistake, I've added the "off command" outside of the "on command" promise.
So it was turning off directly after turning it on .... (this script is launched on motion dectection so when doing my test, there was this launcher activated about every 5min....)

But, in the gladys log, the module Hue dump an error about being "unable to modify the brightness blabla..." when turning off the device, so I probably made a mistake in the understanding of the issue.

I will test again if Hue Module successfully change brightness and turn on the device in //
